### ISSUE TITLE
Remove delay from amp-install-service-worker

### ISSUE THREAD

**Issue by zackargyle** [ampproject/amphtml#12473]
## Service worker does not install until 10 seconds have passed

This delay should be configurable. Ten seconds is a fine default for people who just throw it in, but it is far too long for real use cases.

## How do we reproduce the issue?

[Respective Code](https://github.com/ampproject/amphtml/blob/master/extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js#L104)

**Comment by jridgewell** [comment]
> but it is far too long for real use cases.

Can you expand? This is done so avoid superficial installs (as the [comment](https://github.com/ampproject/amphtml/blob/d4f9694563f50d437cdc53b40501f5dfc9f60ecf/extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js#L98-L101) explains) and also to not block main content on the page with cache priming bs.

**Comment by zackargyle** [comment]
Hey @jridgewell! Thanks for the question. My main issue is that "superficial" is really bias towards something like a news article in this case. Time spent on a page is not the only metric for an engaging site. It biases against a page showing interesting linked content immediately, or the page itself having no links to other interesting content. At Pinterest, for example, our AMP pages are simple landing pages that will link you to our far more engaging PWA, but that PWA loads more slowly because they almost never stay on the AMP page for 12-15s (delay + iframe load + sw install).

**Comment by cramforce** [comment]
@zackargyle What is the time on page you observe on your AMP pages? Have you considered implementing interactive features on the AMP page to make the AMP page more useful? I'm happy to reduce the time, but it should be backed by data for pages that have more use cases than sending users to a page further down.

Implementation wise, I think the best thing would be to switch to loading the SW as a low priority resource instead of the custom timeout.

**Comment by cpapazian** [comment]
>Implementation wise, I think the best thing would be to switch to loading the SW as a low priority resource instead of the custom timeout.

this seems like a perfect solution. let us know if there is anything we can do to help move it along.

>Have you considered implementing interactive features on the AMP page to make the AMP page more useful? I'm happy to reduce the time, but it should be backed by data for pages that have more use cases than sending users to a page further down.

There are a bunch of things we are trying to make our AMP pages more engaging. We added search, follow, and notifications features for authenticated users recently and we have plans to implement a personalized feed of related pins once https://github.com/ampproject/amphtml/issues/11881 lands. 

However, our AMP landing experience is never going to be as engaging as the PWA. In addition to being really hard to deliver a personalized experience without a first-party impression, it just doesn't make sense. Our goal is to make the transition from AMP to PWA as seamless as possible ...

**Comment by cramforce** [comment]
To @ericlindley-g for prioritization in next sprints.

…unless you'd like to send a PR @zackargyle.

**Comment by choumx** [comment]
Bug: `layoutCallback` is probably never called for `layout=nodisplay`. 😓

**Comment by jridgewell** [comment]
Yup... Whoops.
